orders/src/test/java/nl/tudelft/wdm/group1/orders/OrdersApplicationTest.java

```diff
@@ -45,24 +47,28 @@
     public static EmbeddedKafkaRule embeddedKafka = new EmbeddedKafkaRule(1, false, 5, "orders");
 
     private Order defaultOrder;
+    private UUID defaultOrderItemId = UUID.randomUUID();
 
     @Before
-    public void setUp() {
-        defaultOrder = new Order();
+    public void setUp() throws Exception {
+        defaultOrder = new Order(UUID.randomUUID());
+        defaultOrder.addItem(defaultOrderItemId);
         orderRepository.add(defaultOrder);
+        assertThat(orderRepository.find(defaultOrder.getId())).isNotNull();
     }
 
     @Test
     public void createNewOrder() throws Exception {
         MvcResult result = this.mockMvc.perform(
-                post("/orders")
+                post("/orders/" + defaultOrder.getUserId())
         ).andExpect(status().isOk()).andReturn();
 
-        Thread.sleep(2000); // TODO: Remove this ugly hack
+        Thread.sleep(5000); // TODO: Remove this ugly hack
```

Comment:
For some reason this test in particular took more time than others, so I put it at 5s. Ugly, I know but the best I could think of for now.

Reply:
The underlying problem here is that it the message is put through kafka and you don't know when kafka finished right?

Reply:
The embedded kafka instance maybe knows what's going on.

If we really can't know what's going on something like https://github.com/awaitility/awaitility would be an elegant way to test this.

Reply:
Created #12 for this specific thing.